src/ExternalSearch.Providers.GoogleMaps/GoogleMapsExternalSearchProvider.cs

```diff
@@ -456,6 +457,102 @@ public IPreviewImage GetPrimaryEntityPreviewImage(ExecutionContext context, IExt
             return null;
         }
 
+        public ConnectionVerificationResult VerifyConnection(ExecutionContext context, IReadOnlyDictionary<string, object> config)
+        {
+            IDictionary<string, object> configDict = config.ToDictionary(entry => entry.Key, entry => entry.Value);
+            var jobData = new GoogleMapsExternalSearchJobData(configDict);
+            var apiToken = jobData.ApiToken;
+            var client = new RestClient("https://maps.googleapis.com/maps/api");
+            var output = "json";
+            var placeDetailsEndpoint = $"place/details/{output}?";
+            var placeIdEndpoint = $"place/textsearch/{output}?";
+
+            var placeIdRequest = new RestRequest(placeIdEndpoint, Method.GET);
+            placeIdRequest.AddQueryParameter("key", apiToken);
+            placeIdRequest.AddQueryParameter("query", "Google" + " " + "1600 Amphitheatre Parkway, Mountain View, CA 94043.");
+
+            IRestResponse<PlaceIdResponse> placeIdResponse = null;
+            try
+            {
+               placeIdResponse = client.ExecuteAsync<PlaceIdResponse>(placeIdRequest).Result;
+            }
+            catch (Exception exception)
+            {
+                return new ConnectionVerificationResult(false, $"Could not return PlaceIdResponse from Google Maps { exception}");
+            }
+
+            if (!placeIdResponse.IsSuccessful)
+            {
+                return ConstructVerifyConnectionResponse(placeIdResponse);
+            }
+
+            if (placeIdResponse.StatusCode == HttpStatusCode.OK)
+            {
+                var request = new RestRequest(placeDetailsEndpoint, Method.GET);
+                foreach (var placeId in placeIdResponse.Data.Results)
+                {
+                    request.AddParameter("placeid", placeId.PlaceId);
+                    request.AddParameter("key", apiToken);
+                }
+
+                IRestResponse<CompanyDetailsResponse> response = null;
+
+                try
+                {
+                    response = client.ExecuteAsync<CompanyDetailsResponse>(request).Result;
+                }
+                catch (Exception exception)
+                {
+                    return new ConnectionVerificationResult(false, $"Could not return CompanyDetailsResponse from Google Maps. {exception}");
```

Comment:
Update message

`$"Could not fetch company details from Google Maps. {exception.Message}"`

The above is now only putting the exception.Message in the response returned to the user as the user will not want to see a full stack trace. So make sure you also log the full exception

Reply:
updated the message